src/pools/thread/fixed.ts

```diff
@@ -25,6 +25,19 @@ export class FixedThreadPool<
   Data = unknown,
   Response = unknown
 > extends AbstractPool<ThreadWorkerWithMessageChannel, Data, Response> {
+  /**
+   * The tasks map.
+   *
+   * - `key`: The is the message id of each task submitted.
+   * - `value`: An object that contains the worker, the resolve function and the reject function.
+   *
+   * When we receive a message from the worker we get a map entry and resolve/reject the promise based on the message.
+   */
+  protected readonly promiseMap: Map<number, any> = new Map<
```

Comment:
@Shinigami92  just to simplify you, what I did is almost into this JS doc.
When a task is execute we put a record into this map, and when we receive a message from the worker we got the record and resolve/reject the promise.
This is an enhancement because we now avoid to subscribe more and more listeners ( one per task ), and from there I was also thinking to "promote" this on the abstract pool so that cluster pool will take also benefits from this new mechanism